### PR TITLE
HOLD episode downloads chart

### PR DESCRIPTION
**Pull request by sandikbarr**
HOLD #390 showing the episode downloads chart

### CODE REVIEW

File: src/app/core/cms/castle.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        this.replayRoot.next(obj);
+      },
+      (err) => {
+        this.replayRoot.error(err);
+      }
+    );
+  }
+
+  protected getRemote(): HalRemote {
+    return new HalRemote(this.http, Env.CASTLE_HOST);
```
Comment (cavis) [inline src/app/core/cms/castle.service.ts]: So this is assuming no basic-auth required, right?

File: src/app/story/directives/metrics-downloads.component.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+          this.chartData = [new TimeseriesChartModel(dataset, this.story.title, '#61A85D')];
+        } else {
+          this.error = 'This podcast has no download metrics.';
+        }
+      });
+    }
+  }
+
+  checkRequest() {
+    if (!this.story || !this.story.publishedAt) {
```
Comment (cavis) [inline src/app/story/directives/metrics-downloads.component.ts]: I'm still getting an endless spinner, for an unpublished/draft episode.  No console errors, though.  And not making any request to castle.

Comment (sandikbarr) [inline src/app/story/directives/metrics-downloads.component.ts]: ah yes. will fix. the spinner also doesn't go away when you do make a request to castle and it gets a 500, seems like on no data

File: src/app/core/cms/castle.service.ts (hunk 1 above)
Comment (sandikbarr) [inline src/app/core/cms/castle.service.ts]: right. I had a basic auth header stuck into the proxy at one point, but I removed that proxy.

File: src/app/story/directives/metrics-downloads.component.ts (hunk 2 above)
Comment (cavis) [inline src/app/story/directives/metrics-downloads.component.ts]: Oh, okay, so it's actually that my series has no podcast-distribution.

I'm just going to do a quick fix to hide the tab for new-or-podcast-less stories.  And throw up an error message if Castle returns a non-200.

**Comment by gcampo88** [comment]
@cavis your changes look good! Only caveats: I'm getting `The <template> element is deprecated. Use <ng-template> instead`, and currently getting a 406 from castle on the download tab --  `(Phoenix.NotAcceptableError) no supported media type in accept header, expected one of ["json", "hal"]`. Have added a castle ticket (or do we want to just alter that in castle.service?) https://github.com/PRX/castle.prx.org/issues/9. 

Anywho, will +1 this as well but refrain from merging.

**Comment by cavis** [comment]
Away we go!

**Pull request merged**